packages/@aws-cdk/aws-rds/lib/serverless-cluster.ts

```diff
@@ -0,0 +1,544 @@
+
+  private readonly securityGroups: ec2.ISecurityGroup[];
+  private readonly subnetGroup: CfnDBSubnetGroup;
+  private readonly vpc: ec2.IVpc;
+  private readonly vpcSubnets?: ec2.SubnetSelection;
+
+  private readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+  private readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  constructor(scope:Construct, id: string, props: ServerlessClusterProps) {
+    super(scope, id);
+
+    this.vpc = props.vpc ?? new ec2.Vpc(this, 'Vpc', { maxAzs: 2 });
+    this.vpcSubnets = props.vpcSubnets;
+
+    this.singleUserRotationApplication = props.engine.singleUserRotationApplication;
+    this.multiUserRotationApplication = props.engine.multiUserRotationApplication;
+
+    const { subnetIds } = this.vpc.selectSubnets(this.vpcSubnets);
+
+    // Cannot test whether the subnets are in different AZs, but at least we can test the amount.
+    if (subnetIds.length < 2) {
+      Annotations.of(this).addError(`Cluster requires at least 2 subnets, got ${subnetIds.length}`);
+    }
+
+    this.subnetGroup = new CfnDBSubnetGroup(this, 'Subnets', {
+      dbSubnetGroupDescription: `Subnets for ${id} database`,
+      subnetIds,
+    });
+
+    if (props.removalPolicy === RemovalPolicy.RETAIN) {
+      this.subnetGroup.applyRemovalPolicy(RemovalPolicy.RETAIN);
+    }
+
+    this.securityGroups = props.securityGroups ?? [
+      new ec2.SecurityGroup(this, 'SecurityGroup', {
+        description: 'RDS security group',
+        vpc: this.vpc,
+      }),
+    ];
+
+    let secret: DatabaseSecret | undefined;
+    if (!props.masterUser.password) {
+      secret = new DatabaseSecret(this, 'Secret', {
+        username: props.masterUser.username,
+        encryptionKey: props.masterUser.encryptionKey,
+      });
+    }
+
+    // bind the engine to the Cluster
+    const clusterEngineBindConfig = props.engine.bindToCluster(this, {
+      parameterGroup: props.parameterGroup,
+    });
+    const clusterParameterGroup = props.parameterGroup ?? clusterEngineBindConfig.parameterGroup;
+    const clusterParameterGroupConfig = clusterParameterGroup?.bindToCluster({});
+
+    const cluster = new CfnDBCluster(this, 'Resource', {
+      backupRetentionPeriod: props.backupRetention?.toDays(),
+      databaseName: props.defaultDatabaseName,
+      dbClusterIdentifier: props.clusterIdentifier,
+      dbClusterParameterGroupName: clusterParameterGroupConfig?.parameterGroupName,
+      dbSubnetGroupName: this.subnetGroup.ref,
+      deletionProtection: props.deletionProtection,
+      engine: props.engine.engineType,
+      engineVersion: props.engine.engineVersion?.fullVersion,
+      engineMode: 'serverless',
+      enableHttpEndpoint: props.enableHttpEndpoint ?? false,
+      kmsKeyId: props.storageEncryptionKey?.keyArn,
+      masterUsername: secret ? secret.secretValueFromJson('username').toString() : props.masterUser.username,
+      masterUserPassword: secret
```

Comment:
Along with the rename (masterUser -> credentials), take a look at how the instance and cluster are handling these properties now. A bit cleaner on the eyes:

https://github.com/aws/aws-cdk/blob/350105a7f27b6f7fafb5e5dd7f6b3a81a1f5baae/packages/@aws-cdk/aws-rds/lib/cluster.ts#L491-L498

https://github.com/aws/aws-cdk/blob/350105a7f27b6f7fafb5e5dd7f6b3a81a1f5baae/packages/@aws-cdk/aws-rds/lib/cluster.ts#L503-L504

Reply:
nice! aligned to cluster.